TypeScript.YAML-tmLanguage

```diff
@@ -154,15 +154,15 @@ repository:
 
   method-declaration:
     name: meta.method.declaration.ts
-    begin: '\b(?:(abstract)\s+)?\b(?:(public|private|protected)\s+)?\b(?:(async)\s+)?(?:(get|set)\s+)?(?:(new)|(?:\b(constructor)\b)|(?:([a-zA-Z_$][\.\w$]*)\s*(\??)))?\s*(?=\(|\<)'
+    begin: '\b(?:(abstract)\s+)?\b(?:(public|private|protected)\s+)?\b(?:(async)\s+)?(?:(get|set)\s+)?(?:(new)|(?:\b(constructor)\b)|(?:(\[*)(`|'|"*)([a-zA-Z_$][\.\w$]*)(`|'|"*)(\]*)\s*(\??)))?\s*(?=\(|\<)'
```

Comment:
The thing inside the ``|'|"` could be anything, not only `[a-zA-Z_$][\.\w$]*`

For example ``[`123`]``, ``[`@$!$`]`` are both valid method name